salt/modules/win_shortcut.py

```diff
@@ -0,0 +1,471 @@
+            ``Normal``. Valid options are:
+
+            - Normal
+            - Minimized
+            - Maximized
+
+        working_dir (str, optional): The full path to the working directory for
+            the program to run in. This is shown in the ``Start in`` field of
+            the dialog box.
+
+        backup (bool, optional): If there is already a shortcut with the same
+            name, set this value to ``True`` to backup the existing shortcut and
+            continue creating the new shortcut. Default is ``False``
+
+        force (bool, optional): If there is already a shortcut with the same
+            name and you aren't backing up the shortcut, set this value to
+            ``True`` to remove the existing shortcut and create a new with these
+            settings. Default is ``False``
+
+        make_dirs (bool, optional): If the parent directory structure does not
+            exist for the new shortcut, create it. Default is ``False``
+
+        user (str, optional): The user to be the owner of any directories
+            created by setting ``make_dirs`` to ``True``. If no value is passed
+            Salt will use the user account that it is running under. Default is
+            an empty string.
+
+    Returns:
+        bool: True if successful
+
+    Raises:
+        CommandExecutionError: If the path is not a ``.lnk`` or ``.url`` file
+            extension.
+        CommandExecutionError: If there is an existing shortcut with the same
+            name and ``backup`` and ``force`` are both ``False``
+        CommandExecutionError: If the parent directory is not created and
+            ``make_dirs`` is ``False``
+        CommandExecutionError: If there was an error creating the parent
+            directories
+
+    CLI Example:
+
+    .. code-block:: bash
+
+        # Create a shortcut and set the ``Shortcut key`` (``hot_key``)
+        salt * shortcut.create "C:\path\to\shortcut.lnk" "C:\Windows\notepad.exe" hot_key="Ctrl+Alt+N"
+
+        # Create a shortcut and change the icon to the 3rd one in the icon file
+        salt * shortcut.create "C:\path\to\shortcut.lnk" "C:\Windows\notepad.exe" icon_location="C:\path\to\icon.ico" icon_index=2
+
+        # Create a shortcut and change the startup mode to full screen
+        salt * shortcut.create "C:\path\to\shortcut.lnk" "C:\Windows\notepad.exe" window_style="Maximized"
+
+        # Create a shortcut and change the icon
+        salt * shortcut.create "C:\path\to\shortcut.lnk" "C:\Windows\notepad.exe" icon_location="C:\path\to\icon.ico"
+
+        # Create a shortcut and force it to overwrite an existing shortcut
+        salt * shortcut.create "C:\path\to\shortcut.lnk" "C:\Windows\notepad.exe" force=True
+
+        # Create a shortcut and create any parent directories if they are missing
+        salt * shortcut.create "C:\path\to\shortcut.lnk" "C:\Windows\notepad.exe" make_dirs=True
+    """
+    if not path.endswith((".lnk", ".url")):
```

Comment:
can we also verify that the user is passing in a path to target and its what we would expect

Reply:
The path may or may not exist. If it does, it tries to detect that and backup or overwrite depending on other options. Or, maybe I don't understand what you're asking here.

Reply:
We don't need to verify the target is a path that exists currently, but we need to validate the input to ensure someone is passing in a path and nothing else like a command.